src/components/table/column-picker/column-picker.component.ts

```diff
@@ -41,60 +43,174 @@ export class ColumnPickerComponent {
     /** Define a function that return a column move announcement */
     @Input() columnMovedAnnouncement: (column: string, delta: number) => string = this.getColumnMovedAnnouncement;
 
+    /** group settings for customising display */
+    @Input() groupSettings: ReadonlyArray<ColumnPickerGroupSetting> = [];
+
     /** Emits when the selected items change */
     @Output() selectedChange = new EventEmitter<ReadonlyArray<string>>();
 
     /** Emits when the deselected items change */
-    @Output() deselectedChange = new EventEmitter<ReadonlyArray<string>>();
+    @Output() deselectedChange = new EventEmitter<ReadonlyArray<string | ColumnPickerGroupItem>>();
+
+    /** The Nested tree control used for the deselect tree */
+    _treeControl: FlatTreeControl<ColumnPickerTreeNode> = new FlatTreeControl<ColumnPickerTreeNode>(
+        node => node.level,
+        node => node.expandable
+    );
+
+    /** A tree-friendly representation of the deselected data */
+    _treeData: ColumnPickerTreeNode[];
+
+    /** Data source observable bound to the tree control */
+    _treeDataSource: ArrayDataSource<ColumnPickerTreeNode>;
+
+    /** The remaining selectable columns in the deselected list */
+    _availableDeselectedColumns: number = 0;
 
     /** Store the list of deselected columns that can be moved */
-    _deselectedSelection: ReadonlyArray<string> = [];
+    _deselectedSelection: ReadonlyArray<string | ColumnPickerGroupItem> = [];
 
     /** Store the list of selected columns that can be moved */
-    _selectedSelection: ReadonlyArray<string> = [];
+    _selectedSelection: ReadonlyArray<string | ColumnPickerGroupItem> = [];
 
     /** Cache selection during reordering */
-    private _selection: ReadonlyArray<string> = [];
+    private _selection: ReadonlyArray<string | ColumnPickerGroupItem> = [];
 
     /** Get the elements for the selected items */
     @ViewChildren('selectedColumn') selectedElements: QueryList<ElementRef>;
 
     constructor(
         /** Access the LiveAnnounce to provide accessibility on reordering */
-        private _liveAnnouncer: LiveAnnouncer,
+        private readonly _liveAnnouncer: LiveAnnouncer,
         /** We are using OnPush change detection so we must manually trigger CD */
-        private _changeDetectorRef: ChangeDetectorRef
+        private readonly _changeDetectorRef: ChangeDetectorRef
     ) { }
 
+    ngOnChanges(changes: SimpleChanges): void {
+        // recreate tree when deselected changes
+        if (changes.deselected && changes.deselected.currentValue !== changes.deselected.previousValue) {
+            this._rebuildDeselectTree();
+        }
+    }
+
+    /** Parse data into suitable format for the FlatTreeComponent to understand and initialize deselect tree */
+    _rebuildDeselectTree(): void {
```

Comment:
I believe this can be made `private`. Check for the other functions as well, they only need to be public if they are either part of the API of the component, or used directly in the view.